src/queue.c

```diff
@@ -0,0 +1,46 @@
+#include "queue.h"
+
+#ifdef __arm__
+#define MG_MEMORY_BARRIER() asm("dsb; isb;")
+#else
+#define MG_MEMORY_BARRIER() (void) 0
+#endif
+
+// Align given value up
+static unsigned long mg_align(unsigned long value, unsigned long align) {
+  return (value + align - 1) / align * align;
+}
+
```

Comment:
That is correct only for one case: when `align == sizeof(unsigned long)`. Which also must be power of 2.
My formula works for any `align`. For example,  if `align == 7:`: `mg_align(11, 7) == 14`

Reply:
Yes, of course, but you are only aligning to `unsigned long` and that I guess is because of the alignment needed for some DMA controllers. I fail to imagine why you would align to 7 or other non-power-of-2 number